pkg/controller/garbagecollector/garbagecollector_test.go

```diff
@@ -1432,6 +1432,64 @@ func TestConflictingData(t *testing.T) {
 				}),
 			},
 		},
+		{
+			name: "Foreground deletion",
+			steps: []step{
+				// 0,1: setup
+				createObjectInClient("apps", "v1", "deployments", "ns1", makeMetadataObj(deployment1apps)),
+				createObjectInClient("", "v1", "pods", "ns1", makeMetadataObj(pod1ns1, deployment1apps)),
+				// 2,3,4: observe creation
+				processEvent(makeAddEvent(deployment1apps)),
+				processEvent(makeAddEvent(pod1ns1, deployment1apps)),
+				assertState(state{
+					graphNodes:             []*node{makeNode(pod1ns1, withOwners(deployment1apps)), makeNode(deployment1apps)},
+					pendingAttemptToDelete: []*node{},
+				}),
+				// 5,6,7: parent marked for deletion
+				deleteForgroundObjectFromClient("apps", "v1", "deployments", "ns1", "deployment1"),
+				processEvent(makeDeleteForegroundEvent(deployment1apps)),
+				assertState(state{
+					graphNodes:             []*node{makeNode(pod1ns1, withOwners(deployment1apps)), makeNode(deployment1apps)},
+					pendingAttemptToDelete: []*node{makeNode(pod1ns1, withOwners(deployment1apps)), makeNode(deployment1apps)},
+				}),
+				// 8,9: delete child
+				processAttemptToDelete(1),
+				assertState(state{
+					clientActions: []string{
+						"get /v1, Resource=pods ns=ns1 name=podname1",
+						"get apps/v1, Resource=deployments ns=ns1 name=deployment1",
+						"delete /v1, Resource=pods ns=ns1 name=podname1",
+					},
+					graphNodes:             []*node{makeNode(pod1ns1, withOwners(deployment1apps)), makeNode(deployment1apps)},
+					pendingAttemptToDelete: []*node{makeNode(deployment1apps)},
+				}),
+				// 10,11: child deleted
+				processEvent(makeDeleteEvent(pod1ns1, deployment1apps)),
+				assertState(state{
+					graphNodes:             []*node{makeNode(deployment1apps)},
+					pendingAttemptToDelete: []*node{makeNode(deployment1apps)},
+				}),
+				// 12,13: update parent
+				processAttemptToDelete(1),
+				assertState(state{
+					clientActions: []string{
+						"get apps/v1, Resource=deployments ns=ns1 name=deployment1",
+						"get apps/v1, Resource=deployments ns=ns1 name=deployment1",
```

Comment:
would be good to annotate what is happening with these lookups... I can't remember exactly and am a little surprised to see identical sequential gets